boa/src/builtins/typed_array/int8_array.rs

```diff
@@ -0,0 +1,129 @@
+//! This module implements the global `Int8Array` object.
+
+use super::allocate_typed_array;
+use crate::{
+    builtins::typed_array::TypedArrayName, builtins::BuiltIn, object::ConstructorBuilder,
+    property::Attribute, symbol::WellKnownSymbols, value::JsValue, BoaProfiler, Context, JsResult,
+};
+
+/// JavaScript `Array` built-in implementation.
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Int8Array;
+
+impl BuiltIn for Int8Array {
+    const NAME: &'static str = "Int8Array";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, JsValue, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let int8_array = ConstructorBuilder::with_standard_object(
+            context,
+            Self::constructor,
+            context.standard_objects().typed_array_object().clone(),
+        )
+        .name(Self::NAME)
+        .length(Self::LENGTH)
+        .build();
+
+        todo!()
+    }
+}
+
+impl Int8Array {
+    const LENGTH: usize = 3;
+
+    /// <https://tc39.es/ecma262/#sec-typedarray>
+    fn constructor(
+        new_target: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        // 1. If NewTarget is undefined, throw a TypeError exception.
+        if new_target.is_undefined() {
+            return context
+                .throw_type_error("new target was undefined when constructing an Int8Array");
+        }
+
+        // 2. Let constructorName be the String value of the Constructor Name value specified in Table 72 for this TypedArray constructor.
+        let constructor_name = TypedArrayName::Int8Array;
+
+        // 3. Let proto be "%TypedArray.prototype%".
+        let proto = || context.standard_objects().typed_array_object().prototype();
+
+        // 4. Let numberOfArgs be the number of elements in args.
+        let number_of_args = args.len();
+
+        // 5. If numberOfArgs = 0, then
+        if number_of_args == 0 {
+            // a. Return ? AllocateTypedArray(constructorName, NewTarget, proto, 0).
+            return allocate_typed_array(constructor_name, new_target, proto, Some(0), context);
+        }
+        // 6. Else,
+
+        // a. Let firstArgument be args[0].
+        let first_argument = args.get(0).expect("first argument disappeared"); // We already know the legth is not 0
```

Comment:
```suggestion
        let first_argument = args[0];
```
This should already panic with a `index 0 our of bounds` message.